Vitepress theme (#58)

maks-oleksyuk opened the pull request:
- Updated the standard theme
- Added a color theme switcher for the site
- Updated automatic deployment
- Added automatic update of dependencies from dependabot
- Added some plugins to extend functionality (back to top, google analytics, giscus, image viewer, layout view)

Issues that have been identified but not yet fixed:
- Using `{{ }}` to describe the syntax of a twig causes errors. Please use
```
```twig```
```

selwynpolit commented:
> * Updated the standard theme
> * Added a color theme switcher for the site
> * Updated automatic deployment
> * Added automatic update of dependencies from dependabot
> * Added some plugins to extend functionality (back to top, google analytics, giscus, image viewer, layout view)
> 
> Issues that have been identified but not yet fixed:
> 
> * Using `{{ }}` to describe the syntax of a twig causes errors. Please use
> 
> ```
> ```twig```
> ```

This sounds amazing. Thank-you for the hard work! I will review it very soon.  Can you explain the twig issue?  I believe I was using that syntax for twig.  Was I inconsistent?

maks-oleksyuk commented:
During the update, I noticed a problem that when using twig examples in the text (for example, `{{ some_value }}`).

I realized that this is a feature of vue and vitepress, so I converted it to the following ``````twig``````
> ```
> ```twig```
> ```

But with a little patience and searching, I came across [this solution](https://github.com/vuejs/vitepress/discussions/480) that solves this problem.

Therefore please use the following entry for the string twig in text line
`<code v-pre>{{ some_value }}</code>`

selwynpolit commented on book/hooks.md:
```diff
@@ -239,7 +239,7 @@ The render array you want to change will be in the content variable which shows
 Usually fields such as field_image will be automatically rendered by the node template (unless you've tweaked it to display in some other template.)
 
-In your node\'s Twig template you would specify `{{ custom_variable }}` to have it display on every node.
+In your node\'s Twig template you would specify <code v-pre>{{ custom_variable }}</code> to have it display on every node.
```
@maks-oleksyuk I'm curious about this code v-pre stuff. I see in the vitepress docs that we could use:
`This <span v-pre>{{ will be displayed as-is }}</span>`

maks-oleksyuk replied:
Yes, I think we can use this attribute with <span> as well, I tested it and it works, the only difference is that there is no highlighting as when using `

selwynpolit commented on book/attribution.md:
```diff
@@ -29,7 +24,7 @@ Thanks to these rockstars for their contribution to the project.  Please add you
 [Gary Gao](https://github.com/angrytoast) - Query access checking.
 
-[Maks Oleksyuk](https://github.com/maks-oleksyuk) - Amazing work on moving the book to a new theme and many other items.
+[Maks Oleksyuk](https://github.com/maks-oleksyuk) - Amazing work on moving the book to a new theme (x2) and many other items.
```
Nice!

selwynpolit commented:
@maks-oleksyuk I'm not able to build the new theme.  It fails with the following error even though I've tried it several times: 

You have exceeded a secondary rate limit. Please wait a few minutes before you try again. If you reach out to GitHub Support for help, please include the request ID 2D1D:12EC:2BB97FE:5A8C43E:65849723.

![image](https://github.com/selwynpolit/d9book/assets/532848/0c3c654c-ccea-44f4-8a9a-7af5ad335788)

maks-oleksyuk commented:
> You have exceeded a secondary rate limit. Please wai

@selwynpolit I apologize for this, I completely forgot about the limits, because I was checking on my own repository which does not have many deployments, for now I left this action as a comment, maybe it will be fixed in the future

selwynpolit commented:
Thanks for fixing that @maks-oleksyuk - I did open a support ticket with Github - maybe they will have a useful response.  I did find one problem.  The nodes_n_fields.md link on the sidebar gives a 404.  I see that you added  `permalink: /nodes-and-fields` at the top of the file.  Should I just rename the file to nodes-and-fields.md?  I looked in the docs for a sidebar and found [this](https://vitepress.dev/reference/default-theme-sidebar#sidebar) where they talk about creating a themeConfig.sidebar file but I don't see that in the repo. I'm guessing that means that the sidebar is dynamically generated from the markdown files it finds at build time.

selwynpolit commented:
hmm. I also found a problem with the getting off the island page.  Renaming it and removing the `rewrite` in config.mjs seemed to cure that.  This theme is really nice!

selwynpolit commented:
@maks-oleksyuk any idea how we can reinstate the date updated functionality that we had in the last theme?  I found that useful.

selwynpolit commented:
@maks-oleksyuk Could you take a look and see if there is some reason I can't get the project to build?  I made some changes to the upgrade.md file and it keeps failing.

maks-oleksyuk commented:
Hi @selwynpolit, I think this is due to the fact that the **Upload artifact** action loads an artifact with the same name from several builds at once. There was a recent update for these actions with a change in their architecture, I will try to update them and fix it.

To fix this, you need to delete the artifacts you made and restart these actions.

Will it be possible for me to make multiple commits directly to `gh-pages` to update the workflow?

selwynpolit commented:
ah.  I'm not sure how to delete the artifacts but yes, please go ahead and update the workflow.

maks-oleksyuk commented:
@selwynpolit I updated the workflow and now it shouldn't give any errors with artifacts. I also fixed the automatic removal of old deployments, so now only 1 active environment will be displayed on github, not 500+ as it was before

selwynpolit commented:
thanks @maks-oleksyuk
